lib/saluki-components/src/destinations/datadog_events_service_checks/mod.rs

```diff
@@ -0,0 +1,234 @@
+    ///
+    /// Defaults to unset.
+    #[serde(default)]
+    dd_url: Option<String>,
+}
+
+impl DatadogEventsServiceChecksConfiguration {
+    /// Creates a new `DatadogEventsServieCheckConfiguration` from the given configuration.
+    pub fn from_configuration(config: &GenericConfiguration) -> Result<Self, GenericError> {
+        Ok(config.as_typed()?)
+    }
+
+    fn api_base(&self) -> Result<Uri, GenericError> {
+        match &self.dd_url {
+            Some(url) => Uri::try_from(url).map_err(Into::into),
+            None => {
+                let site = if self.site.is_empty() {
+                    DEFAULT_SITE
+                } else {
+                    self.site.as_str()
+                };
+                let authority = format!("api.{}", site);
+
+                Uri::builder()
+                    .scheme("https")
+                    .authority(authority.as_str())
+                    .path_and_query("/")
+                    .build()
+                    .map_err(Into::into)
+            }
+        }
+    }
+}
+
+#[async_trait]
+impl DestinationBuilder for DatadogEventsServiceChecksConfiguration {
+    fn input_data_type(&self) -> DataType {
+        DataType::EventD | DataType::ServiceCheck
+    }
+
+    async fn build(&self) -> Result<Box<dyn Destination + Send>, GenericError> {
+        let http_client = HttpClient::https()?;
+
+        let api_base = self.api_base()?;
+
+        let events_request_builder = RequestBuilder::new(
+            self.api_key.clone(),
+            api_base.clone(),
+            EventsServiceChecksEndpoint::Events,
+        )?;
+        let service_checks_request_builder = RequestBuilder::new(
+            self.api_key.clone(),
+            api_base,
+            EventsServiceChecksEndpoint::ServiceChecks,
+        )?;
+        Ok(Box::new(DatadogEventsServiceChecks {
+            http_client,
+            events_request_builder,
+            service_checks_request_builder,
+        }))
+    }
+}
+
+impl MemoryBounds for DatadogEventsServiceChecksConfiguration {
+    fn specify_bounds(&self, _builder: &mut MemoryBoundsBuilder) {}
+}
+
+pub struct DatadogEventsServiceChecks {
+    http_client: HttpClient<HttpsConnector<HttpConnector>, String>,
+    events_request_builder: RequestBuilder,
+    service_checks_request_builder: RequestBuilder,
+}
+
+#[async_trait]
+impl Destination for DatadogEventsServiceChecks {
+    async fn run(mut self: Box<Self>, mut context: DestinationContext) -> Result<(), ()> {
+        let Self {
+            http_client,
+            mut events_request_builder,
+            mut service_checks_request_builder,
+        } = *self;
+
+        // Spawn our IO task to handle sending requests.
+        let (io_shutdown_tx, io_shutdown_rx) = oneshot::channel();
+        let (requests_tx, requests_rx) = mpsc::channel(32);
+        // let events_service_checks = Metrics::from_component_context(context.component_context());
```

Comment:
Seems like this line should get removed?

Reply:
Oops, good catch